docs/source-pytorch/common/checkpointing_expert.rst

```diff
@@ -8,6 +8,13 @@ Checkpointing (expert)
 
 TODO: I don't understand this...
 
+*********************************
+Writing your own Checkpoint class
+*********************************
+
+We provide ``Checkpoint`` class, for easier subclassing. Users may want to subclass this class in case of writing custom ``ModelCheckpoint`` callback, so that the ``Trainer`` recognizes the custom class as a checkpointing callback.
```

Comment:
I don't think this fits here, because this page is about the CheckpointIO plugin and we don't want to confuse the two concepts I think. 

> so that the ``Trainer`` recognizes the custom class as a checkpointing callback.

This is not something the user cares about, this is just an implementation detail.

I propose to write the section with the main point being the recommendation of using the Checkpoint as the base class for highly custom checkpointing callbacks, and also provide a very simple example.

Reply:
But, while I'm defending this particular sentence, I fully agree that the section could get a rewrite and maybe moved somewhere else :)